internal/controller/environmentrequest_controller.go

```diff
@@ -116,19 +116,25 @@ func (r *EnvironmentRequestReconciler) reconcile(ctx context.Context, environmen
 	logger.V(1).Info("environment provider count", "active", len(environmentProviders.activeJobs), "successful", len(environmentProviders.successfulJobs), "failed", len(environmentProviders.failedJobs))
 
 	// Check providers availability
-	// TODO: Update status
 	providers := etosv1alpha1.Providers{
 		IUT:            environmentrequest.Spec.Providers.IUT.ID,
 		ExecutionSpace: environmentrequest.Spec.Providers.ExecutionSpace.ID,
 		LogArea:        environmentrequest.Spec.Providers.LogArea.ID,
 	}
 	if err := checkProviders(ctx, r, environmentrequest.Namespace, providers); err != nil {
+		meta.SetStatusCondition(&environmentrequest.Status.Conditions, metav1.Condition{Status: metav1.ConditionTrue, Type: StatusFailed, Message: "Provider check failed", Reason: err.Error()})
```

Comment:
`StatusReady` is the one that we use in the environment request and that is the status that the testrun controller uses to determine if the environmentrequest has completed.

Reply:
Changed to ConditionFalse, StatusReady